components/script/script_thread.rs

```diff
@@ -2089,6 +2105,39 @@ impl ScriptThread {
             },
             _ => (),
         }
+    }
+
+    fn handle_webdriver_msg(
+        &self,
+        pipeline_id: PipelineId,
+        msg: WebDriverScriptCommand,
+        can_gc: CanGc,
+    ) {
+        // https://github.com/servo/servo/issues/23535
+        // These two messages need different treatment since the JS script might mutate
+        // `self.documents`, which would conflict with the immutable borrow of it that
+        // occurs for the rest of the messages
+        match msg {
+            WebDriverScriptCommand::ExecuteScript(ref _script, ref _reply) => {
+                return self.pending_webdriver_script_events.borrow_mut().push(
+                    WebdriverScriptEvent {
+                        pipeline_id,
+                        msg,
+                        can_gc,
+                    },
+                );
+            },
+            WebDriverScriptCommand::ExecuteAsyncScript(ref _script, ref _reply) => {
+                return self.pending_webdriver_script_events.borrow_mut().push(
+                    WebdriverScriptEvent {
+                        pipeline_id,
+                        msg,
+                        can_gc,
+                    },
+                );
+            },
```

Comment:
Since they are same, you can merge into one.
```suggestion
            WebDriverScriptCommand::ExecuteScript(..) | WebDriverScriptCommand::ExecuteAsyncScript(..) => {
                return self.pending_webdriver_script_events.borrow_mut().push(
                    WebdriverScriptEvent {
                        pipeline_id,
                        msg,
                        can_gc,
                    },
                );
            },
```

Reply:
Thanks. Let me update.